pulsar-io/influxdb/src/main/java/org/apache/pulsar/io/influxdb/InfluxDBAbstractSink.java

```diff
@@ -0,0 +1,156 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pulsar.io.influxdb;
+
+import com.google.common.collect.Lists;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.collections.CollectionUtils;
+import org.apache.pulsar.functions.api.Record;
+import org.apache.pulsar.io.core.Sink;
+import org.apache.pulsar.io.core.SinkContext;
+import org.influxdb.InfluxDB;
+import org.influxdb.dto.BatchPoints;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * A simple abstract class for InfluxDB sink
+ */
+@Slf4j
+public abstract class InfluxDBAbstractSink<T> implements Sink<T> {
+
+    private InfluxDBSinkConfig influxDBSinkConfig;
+    private InfluxDB influxDB;
+    private InfluxDB.ConsistencyLevel consistencyLevel;
+    private String influxDatabase;
+    private String retentionPolicy;
+
+    protected InfluxDBBuilder influxDBBuilder = new InfluxDBBuilderImpl();
+
+    private long batchTimeMs;
+    private int batchSize;
+    private List<Record<T>> incomingList;
+    private ScheduledExecutorService flushExecutor;
+
+    @Override
+    public void open(Map<String, Object> config, SinkContext sinkContext) throws Exception {
+        influxDBSinkConfig = InfluxDBSinkConfig.load(config);
+        influxDBSinkConfig.validate();
+
+        try {
+            consistencyLevel = InfluxDB.ConsistencyLevel.valueOf(influxDBSinkConfig.getConsistencyLevel().toUpperCase());
+        } catch (IllegalArgumentException e) {
+            throw new IllegalArgumentException("Illegal Consistency Level, valid values are: "
+                + Arrays.asList(InfluxDB.ConsistencyLevel.values()));
+        }
+
+        influxDatabase = influxDBSinkConfig.getDatabase();
+        retentionPolicy = influxDBSinkConfig.getRetentionPolicy();
+
+        influxDB = influxDBBuilder.build(influxDBSinkConfig);
+
+        // create the database if not exists
+        List<String> databases = influxDB.describeDatabases();
+        if (!databases.contains(influxDatabase)) {
+            influxDB.createDatabase(influxDatabase);
```

Comment:
a little suggestion：
```java
if (!databases.contains(influxDatabase)) {
    if(influxDBSinkConfig.isCreateDatabase()) {
         influxDB.createDatabase(influxDatabase);
      }
      else {
        throw new IllegalArgumentException("This " + influxDBSinkConfig.getDatabase() + " database does not exist!");
      }
}
```

Reply:
@ambition119 Thanks for the reviewing, it's not necessary to do this as field `database` is required and has been checked in method `validate()`.